### PR TITLE
ROC-5438 DQ: expert report page

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-5438

### Change description
Add 'Have you already got a report written by an expert?' page
Currently based on https://github.com/hmcts/cmc-citizen-frontend/pull/1627 until that is merged to master.

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [x] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/directions-questionnaire/forms/models/selfWitness.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class SelfWitness implements CompletableTask {
 
   deserialize (input?: any): SelfWitness {
     if (input && input.option) {
-      this.option = YesNoOption.fromObject(input.option)
+      this.option = YesNoOption.fromObject(input.option) || YesNoOption.fromObject(input.option.option)
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/forms/models/selfWitness.ts]: this doesn't seem right. looks like something needs renaming or returned differently

File: src/main/features/directions-questionnaire/routes/expert-reports.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        renderView(form, res)
+      } else {
+        const claim: Claim = res.locals.claim
+        const draft: Draft<DirectionsQuestionnaireDraft> = res.locals.draft
+        const user: User = res.locals.user
+
+        form.model.removeExcessRows()
+        draft.document.expertReports = form.model
+
+        const declared: boolean = draft.document.expertReports.declared
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/routes/expert-reports.ts]: why the explicit const declaration? can't it just be used directly in the if statement?

File: src/main/features/directions-questionnaire/views/expert-reports.njk [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+      <form method="post" novalidate
+            class="analytics-submit-event-trigger"
+            data-event-action="DQ: Have you already got a report written by an expert?"
+            data-event-label-from="declared">
+
+        {{ csrfProtection(csrf) }}
+
+        {% set errorHasAnyExpertReports = form.errorFor('declared') %}
+        <div class="statement-of-means-container">
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/views/expert-reports.njk]: this class needs to be renamed either here or in the scss file to be more generic

File: src/main/features/directions-questionnaire/routes/expert-reports.ts (hunk 2 above)
Comment (andrewwa-kainos) [reply src/main/features/directions-questionnaire/routes/expert-reports.ts]: Forward-looking. The next story will also have logic based on this about where to redirect to.

File: src/main/features/directions-questionnaire/views/expert-reports.njk (hunk 3 above)
Comment (andrewwa-kainos) [reply src/main/features/directions-questionnaire/views/expert-reports.njk]: The div itself seems unnecessary here. Will remove.
(Copy and paste misdemeanour of course 👎 )

File: src/main/modules/nunjucks/index.ts [hunk 4]
```diff
@@ -178,6 +178,8 @@ export class Nunjucks {
     nunjucksEnv.addGlobal('PartyType', PartyType)
     nunjucksEnv.addGlobal('IncomeExpenseSchedule', IncomeExpenseSchedule)
     nunjucksEnv.addGlobal('FreeMediationOption', FreeMediationOption)
+
+    nunjucksEnv.addFilter('json', (o) => JSON.stringify(o))
```
Comment (andrewwa-kainos) [inline src/main/modules/nunjucks/index.ts]: Committed by mistake

**Pull request closed without merging**